packages/angular-cli/commands/generate.ts

```diff
@@ -56,6 +57,17 @@ function mapBlueprintName(name: string): string {
   return mappedName ? mappedName : name;
 }
 
+function mapSufix(args: any): any {
+  for (let alias in aliasMap) {
```

Comment:
This for-loop is not useful; just do an `if (!(args[0].toLowerCase() in aliasMap) { return args[1]; }`. There's only 1 case at  a time.

Basically this whole function can be simplified to

```typescript
function mapSuffix(blueprint: string, name: string) : string {
  name.replace(new RegExp(`-?${blueprint}$`, 'i'), '');
}
```

And if you add a simple comment explain that this removes the suffix from the name, you can inline it above instead of making it a function call.